catalogue/webapp/services/wellcome/catalogue/images.ts

```diff
@@ -29,6 +29,30 @@ type GetImageProps = {
   include?: ImageInclude[];
 };
 
+/** Creates the YYYY-MM-DD date string we pass to the API.
+ *
+ * Note: the filter GUI expects users to enter dates as a four-digit year (e.g. 1939).
+ * We pin to the start/end of the year so that the range is inclusive.
+ *
+ * e.g. a user who searches for works 'to 2001' should find works created in 2001.
+ */
+function toIsoDateString(
+  s: string | undefined,
+  range: 'to' | 'from'
+): string | undefined {
+  if (s) {
+    try {
+      const d = new Date(s);
+      const year = d.getUTCFullYear().toString().padStart(4, '0');
+
+      return range === 'from' ? `${year}-01-01` : `${year}-12-31`;
+    } catch (e) {
+      return undefined;
+    }
+  }
+  return undefined;
+}
+
```

Comment:
Probably preferable in this instance, rather than copying and pasting, to move the existing implementation of this out of `works` and into a common file (the existing `catalogue/index` would be fine I think) to be imported in both `works` and `images`

Reply:
Definitely agreed - will wheel it in there now